### PR TITLE
LIbrary packaging

### PR DESCRIPTION
**Pull request by voodoo11**
- Package now contains only the library code
- Examples are extracted to separate directory
- Dependencies are splitted into main, local and examples
- Poetry can be used to easily run examples

### CODE REVIEW

File: examples/examples.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
```
Comment (janchorowski) [inline examples/examples.py:1]:
```suggestion
#!/usr/bin/env python
```
This is the main entrypoint for running all examples with centralized configuration.
However, all examples are self0contained and can also be run directly.
```

File: README.md [hunk 2]
```diff
@@ -148,22 +139,27 @@ When the process is complete, the App will be up and running inside a Docker con
...
-- **Virtual Python Environment:** Create a new environment and install the required packages to isolate the dependencies of this project from your system's Python:
+- **Install poetry:**
 
     ```bash
-    # Creates an env called pw-env and activates this environment.
-    python -m venv pw-env && source pw-env/bin/activate
+    pip install poetry
+    ```
+
+- **Install dependencies:**
```
Comment (voodoo11) [inline README.md]:
```suggestion
- **Install llm_app and dependencies:** 
```

File: README.md [hunk 3]
```diff
@@ -148,22 +139,27 @@ When the process is complete, the App will be up and running inside a Docker con
...
+
+- **Install llm_app and dependencies:** 
 
-    pip install --upgrade -r requirements.txt
+    ```bash
+    poetry install --with examples --extra local
     ```
 
-- **Run the App:** You can start the application with the command:
+    You can ommit `--extra local` part if you're not going to run local example.
```
Comment (voodoo11) [inline README.md:154]:
```suggestion
    You can ommit `--extras local` part if you're not going to run local example.
```

File: README.md [hunk 4]
```diff
@@ -148,22 +139,27 @@ When the process is complete, the App will be up and running inside a Docker con
...
-    # Creates an env called pw-env and activates this environment.
-    python -m venv pw-env && source pw-env/bin/activate
+    pip install poetry
+    ```
+
+- **Install llm_app and dependencies:** 
 
-    pip install --upgrade -r requirements.txt
+    ```bash
+    poetry install --with examples --extra local
```
Comment (voodoo11) [inline README.md:151]:
```suggestion
    poetry install --with examples --extras local
```

File: README.md [hunk 5]
```diff
@@ -148,22 +139,27 @@ When the process is complete, the App will be up and running inside a Docker con
...
+
+- **Install llm_app and dependencies:** 
 
-    pip install --upgrade -r requirements.txt
+    ```bash
+    poetry install --with examples --extras local
     ```
 
-- **Run the App:** You can start the application with the command:
+    You can ommit `--extras local` part if you're not going to run local example.
```
Comment (janchorowski) [inline README.md:154]:
```suggestion
    You can ommit `--extras local` part if you're only going to use external LLM APIs and not going to run local example.
```

**Review by janchorowski — APPROVED** [review]

**Pull request merged**